src/ngraph/op/experimental/dyn_broadcast.cpp

```diff
@@ -134,3 +143,17 @@ void op::DynBroadcast::generate_adjoints(autodiff::Adjoints& adjoints, const Nod
 {
     throw ngraph_error("generate_adjoints not implemented for DynBroadcast");
 }
+
+void ngraph::op::DynBroadcast::add_default_broadcast_axes()
+{
+    NGRAPH_CHECK(get_input_partial_shape(0).rank().is_static() &&
+                     get_input_partial_shape(1)[0].is_static(),
+                 "Optional parameter broadcast_axes is supported only if rank of input and target "
+                 "shape are static.");
```

Comment:
You cannot relay on dynamic/static shape here, because operation should remain reshapable regardless of the current input nodes. This code should work regardless of input shapes. That means that there should be a sub-graph with explicit ShapeOf and other ops are added to compute this thread input. Or, simpler and probably better alternative, absence of the third input should be treated as a separate case where Python rules of broadcasting is applied and 3rd port remains unconnected. Anyway, after introduction of versioning mechanism we have more freedom in this op modification and probably it is better to change 3rd input semantic to make it more simple and intuitive.